sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/reflect/DoFnInvokersTest.java

```diff
@@ -199,302 +110,222 @@ public void testDoFnInvokersReused() throws Exception {
+    invokeProcessElement(fn);
+    assertTrue(DoFnInvokersTestHelper.wasStaticAnonymousDoFnInvoked(fn));
   }
 
+  // ---------------------------------------------------------------------------------------
+  // Tests for wrapping exceptions.
+  // ---------------------------------------------------------------------------------------
+
   @Test
   public void testProcessElementException() throws Exception {
-    DoFn<Integer, Integer> fn =
-        new DoFn<Integer, Integer>() {
-          @ProcessElement
-          public void processElement(@SuppressWarnings("unused") ProcessContext c) {
-            throw new IllegalArgumentException("bogus");
-          }
-        };
-
     thrown.expect(UserCodeException.class);
     thrown.expectMessage("bogus");
-    DoFnInvokers.INSTANCE.newByteBuddyInvoker(fn).invokeProcessElement(null, null);
+    DoFnInvokers.INSTANCE
+        .newByteBuddyInvoker(
+            new DoFn<Integer, Integer>() {
+              @ProcessElement
+              public void processElement(@SuppressWarnings("unused") ProcessContext c) {
+                throw new IllegalArgumentException("bogus");
+              }
+            })
+        .invokeProcessElement(null, null);
   }
 
   @Test
   public void testStartBundleException() throws Exception {
-    DoFn<Integer, Integer> fn =
-        new DoFn<Integer, Integer>() {
-          @StartBundle
-          public void startBundle(@SuppressWarnings("unused") Context c) {
-            throw new IllegalArgumentException("bogus");
-          }
-
-          @ProcessElement
-          public void processElement(@SuppressWarnings("unused") ProcessContext c) {}
-        };
-
     thrown.expect(UserCodeException.class);
     thrown.expectMessage("bogus");
-    DoFnInvokers.INSTANCE.newByteBuddyInvoker(fn).invokeStartBundle(null);
+    DoFnInvokers.INSTANCE
+        .newByteBuddyInvoker(
+            new DoFn<Integer, Integer>() {
+              @StartBundle
+              public void startBundle(@SuppressWarnings("unused") Context c) {
+                throw new IllegalArgumentException("bogus");
+              }
+
+              @ProcessElement
+              public void processElement(@SuppressWarnings("unused") ProcessContext c) {}
+            })
+        .invokeStartBundle(null);
   }
 
   @Test
   public void testFinishBundleException() throws Exception {
-    DoFn<Integer, Integer> fn =
-        new DoFn<Integer, Integer>() {
-          @FinishBundle
-          public void finishBundle(@SuppressWarnings("unused") Context c) {
-            throw new IllegalArgumentException("bogus");
-          }
-
-          @ProcessElement
-          public void processElement(@SuppressWarnings("unused") ProcessContext c) {}
-        };
-
     thrown.expect(UserCodeException.class);
     thrown.expectMessage("bogus");
-    DoFnInvokers.INSTANCE.newByteBuddyInvoker(fn).invokeFinishBundle(null);
+
+    DoFnInvokers.INSTANCE
```

Comment:
For the various exception tests, it would be good to separate the `.newByteBuddyInvoker` call (do it before the `thrown.expect`) from the `invoke...` call (do it after the `thrown.expect`).